File: DKM_js/src/xrplClient.js
const xrpl = require('xrpl');

async function connectClient() {
  const client = new xrpl.Client('wss://s.altnet.rippletest.net:51233');
  await client.connect();
  console.log("Connected to XRPL");
  return client;
}

async function disconnectClient(client) {
  await client.disconnect();
  console.log("Disconnected from XRPL");
}

async function createFundedAccount(client) {
  const wallet = xrpl.Wallet.generate();
  await client.fundWallet(wallet);

  console.log(`Funded account: ${wallet.address}`);
  return wallet;
}

module.exports = { connectClient, disconnectClient, createFundedAccount };

File: nodes/src/main/java/com/example/XRPLClient.java
package com.example;

import org.xrpl.xrpl4j.client.JsonRpcClient;
import org.xrpl.xrpl4j.client.XrplClient;
import org.xrpl.xrpl4j.wallet.Wallet;
import org.xrpl.xrpl4j.wallet.WalletFactory;

import java.net.URI;

public class XRPLClient {
    private XrplClient client;

    public XRPLClient() throws Exception {
        client = new XrplClient(URI.create("https://s.altnet.rippletest.net:51234"));
    }

    public Wallet createFundedAccount() throws Exception {
        Wallet wallet = WalletFactory.getInstance().randomWallet(true).wallet();
        client.fundAccount(wallet.classicAddress());
        System.out.println("Funded account: " + wallet.classicAddress());
        return wallet;
    }

    public XrplClient getClient() {
        return client;
    }
}
